distributedcombigrid/src/sgpp/distributedcombigrid/CMakeLists.txt

```diff
@@ -0,0 +1,294 @@
+    set(Boost_NO_SYSTEM_PATHS ON)
+    find_package(Boost ${Boost_VERSION} REQUIRED COMPONENTS serialization program_options system unit_test_framework)
+
+    list(POP_BACK CMAKE_MESSAGE_INDENT)
+endif ()
+
+target_include_directories(discotec PRIVATE Boost::headers)
+target_link_libraries(discotec PRIVATE Boost::serialization Boost::program_options Boost::system )
+
+message(STATUS "Fetching glpk")
+FetchContent_Declare(
+        glpk
+        URL http://ftp.gnu.org/gnu/glpk/glpk-4.55.tar.gz # TODO: use newer and not hardcoded Version
+        QUIET
+)
+FetchContent_MakeAvailable(glpk)
+message(STATUS "GLPK: configure ...")
+execute_process(COMMAND ${glpk_SOURCE_DIR}/configure --prefix=${glpk_BINARY_DIR} OUTPUT_QUIET ERROR_QUIET COMMAND_ERROR_IS_FATAL ANY)
+message(STATUS "GLPK: build ...")
+execute_process(COMMAND make -j ${nproc} OUTPUT_QUIET ERROR_QUIET COMMAND_ERROR_IS_FATAL ANY)
+message(STATUS "GLPK: install ...")
+execute_process(COMMAND make install OUTPUT_QUIET COMMAND_ERROR_IS_FATAL ANY)
+set_property(DIRECTORY APPEND PROPERTY CMAKE_CONFIGURE_DEPENDS ${glpk_BINARY_DIR})
+target_include_directories(discotec PUBLIC ${glpk_BINARY_DIR}/include)
+target_link_libraries(discotec PUBLIC ${glpk_BINARY_DIR}/lib/libglpk.a)
+
+
+if (DISCOTEC_HDF5)
+    enable_language(C)
+
+    set(HDF5_PREFER_PARALLEL ON)
+    find_package(HDF5) # TODO: add minimum version and components
+
+
+    if (HDF5_FOUND AND HDF5_IS_PARALLEL)
+        message(STATUS "Found HDF5")
+    else ()
+        if (NOT DISCOTEC_BUILD_MISSING_DEPS)
+            message(FATAL_ERROR "HDF5 not found")
+        endif ()
+        list(APPEND CMAKE_MESSAGE_INDENT "HDF5: ")
+
+        message(WARNING "Parallel HDF5 not found, building own HDF5!")
+        FetchContent_Declare(
+                HDF5
+                GIT_REPOSITORY git@github.com:HDFGroup/hdf5.git
+                GIT_TAG hdf5-1_13_0
+                GIT_SHALLOW TRUE
+                QUIET
+                OVERRIDE_FIND_PACKAGE
+        )
+        set(HDF5_ENABLE_PARALLEL ON CACHE BOOL "" FORCE)
+        set(HDF5_IS_PARALLEL ON CACHE BOOL "" FORCE)
+        set(HDF5_EXTERNALLY_CONFIGURED ON CACHE BOOL "" FORCE)
+        set(HDF5_EXTERNALLY_CONFIGURED ON CACHE BOOL "" FORCE)
+        set(HDF5_BUILD_PARALLEL_TOOLS OFF CACHE BOOL "" FORCE)
+        set(BUILD_SHARED_LIBS OFF CACHE BOOL "" FORCE)
+        set(HDF5_BUILD_FRAMEWORKS OFF CACHE BOOL "" FORCE)
+        set(BUILD_TESTING OFF CACHE BOOL "" FORCE)
+        set(HDF5_BUILD_TOOLS OFF CACHE BOOL "" FORCE)
+        set(TEST_SHELL_SCRIPTS OFF CACHE BOOL "" FORCE)
+        set(HDF5_TEST_TOOLS OFF CACHE BOOL "" FORCE)
+        set(HDF5_TEST_EXAMPLES OFF CACHE BOOL "" FORCE)
+        set(HDF5_TEST_SWMR OFF CACHE BOOL "" FORCE)
+        set(HDF5_TESTS_PARALLEL OFF CACHE BOOL "" FORCE)
+        set(HDF5_BUILD_EXAMPLES OFF CACHE BOOL "" FORCE)
+        set(HDF5_BUILD_CPP_LIB OFF CACHE BOOL "" FORCE)
+        set(HDF5_BUILD_FORTRAN OFF CACHE BOOL "" FORCE)
```

Comment:
hm, DisCoTec uses HDF5 only through HighFive. But! If we use selalib, that uses HDF5 directly and requires the Fortran bindings.

Reply:
I added a TODO, so we don't forget enabling it.